RecoEgamma/EgammaElectronProducers/plugins/alpaka/PixelMatchingAlgo.dev.cc

```diff
@@ -0,0 +1,308 @@
+#include "RecoEgamma/EgammaElectronAlgos/interface/alpaka/ftsFromVertexToPointPortable.h"
+#include "RecoEgamma/EgammaElectronAlgos/interface/alpaka/helixArbitraryPlaneCrossing.h"
+#include "RecoEgamma/EgammaElectronAlgos/interface/alpaka/helixForwardPlaneCrossing.h"
+
+#include "PixelMatchingAlgo.h"
+#include "DataFormats/EgammaReco/interface/alpaka/EleRelPointPairPortable.h"
+#include "DataFormats/EgammaReco/interface/alpaka/Plane.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+        template<typename TAcc, typename T>
+	ALPAKA_FN_ACC ALPAKA_FN_INLINE T getZVtxFromExtrapolation(TAcc const& acc,
+								  const Vec3d& primeVtxPos,
+								  const Vec3d& hitPos,
+								  const Vec3d& candPos) {
+		auto sq = [](T x) { return x * x; };
+
+		auto calRDiff2 = [sq](const Vec3d& p1, const Vec3d& p2) {
+		  return sq(p2[0] - p1[0]) + sq(p2[1] - p1[1]);
+		};
+		const T r1Diff = alpaka::math::sqrt(acc, calRDiff2(primeVtxPos, hitPos));
+		const T r2Diff = alpaka::math::sqrt(acc, calRDiff2(hitPos, candPos));
+
+		T zvtx = hitPos[2] - r1Diff * (candPos[2] - hitPos[2]) / r2Diff;
+		return zvtx;
+	}
+
+	template<typename TAcc, typename T>
+	ALPAKA_FN_ACC ALPAKA_FN_INLINE T getCutValue(TAcc const& acc, const T et, const T highEt, const T highEtThres, const T lowEtGrad) {
+		return highEt + alpaka::math::min(acc, static_cast<T>(0.), et - highEtThres) * lowEtGrad;
+	}
+
+	//--- Kernel for printing the SC SoA
+	class printSCSoA {
+	public:
+	template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+	ALPAKA_FN_ACC void operator()(TAcc const& acc,
+									reco::SuperclusterDeviceCollection::View view,
+									int32_t size) const 
+		{
+#if 0
+			//const int32_t thread = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u];
+			for (int32_t i : uniform_elements(acc, size)) 
+			{
+
+				if(i>=size)
+				    break;
+
+				auto sc = view[i];
+
+				printf("For SC i=%d Energy is :%f , theta is :%f , and r is %lf and phi is %lf,  \n",i,sc.scEnergy(),sc.scSeedTheta(), sc.scR(),sc.scPhi() ) ;
+				float x = sc.scR() * alpaka::math::sin(acc,sc.scSeedTheta()) * alpaka::math::cos(acc,sc.scPhi());
+				float y = sc.scR() * alpaka::math::sin(acc,sc.scSeedTheta()) * alpaka::math::sin(acc,sc.scPhi());
+				float z = sc.scR() * alpaka::math::cos(acc,sc.scSeedTheta());
+				printf("x: %lf,  y: %lf,  z %lf ",x,z,y);
+				Vector3d position{x,y,z};
+				printf("  Value of perp2 %lf \n",x*x+y*y);
+				printf("Calculate the magnetic field with the parabolic approximation at the SC position : %f\n",magneticFieldParabolicPortable::magneticFieldAtPoint(acc, position));			
+			}
+#endif
+		}
+	};
+
+   	//--- Kernel for printing the electron seeds SoA
+	class printElectronSeedSoA {
+	public:
+	template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+	ALPAKA_FN_ACC void operator()(TAcc const& acc,
+									reco::EleSeedDeviceCollection::View view,
```

Comment:
```suggestion
									reco::EleSeedDeviceCollection::ConstView view,
```
because the `view` is not modified.